web-common/src/features/dashboards/time-dimension-details/charts/utils.ts

```diff
@@ -112,6 +139,44 @@ export function sanitizeSpecForTDD(
   const timeUnit = timeGrainToVegaTimeUnitMap[timeGrain];
   xEncoding.timeUnit = timeUnit;
 
+  if (isTimeComparison) {
+    yEncoding.field = "measure";
+
+    sanitizedSpec.transform = [
+      { timeUnit: timeUnit, field: "comparison\\.ts", as: "comparison_ts" },
+      { fold: ["ts", "comparison_ts"] },
+      {
+        calculate: `(datum['key'] === 'comparison_ts' ? datum['comparison.${measureName}'] : datum['${measureName}'])`,
```

Comment:
**Security Issue**: The dynamic construction of the calculation string using template literals can lead to injection vulnerabilities if not properly sanitized. <br> **Fix**: Ensure that any dynamic values used in the calculation are sanitized or validated to prevent injection attacks. <br> **Code Suggestion**: 
 ```
 -        calculate: '(datum['key'] === 'comparison_ts' ? datum['comparison.${measureName}'] : datum['${measureName}'])',
 +        calculate: '(datum['key'] === 'comparison_ts' ? sanitize(datum['comparison.${measureName}']) : sanitize(datum['${measureName}']))',
 ```